### PR TITLE
Added ND nearest interpolation (batched).

### PR DESCRIPTION
**Pull request by kevinjohncutler**

### CODE REVIEW

**Comment by ryanirl** [comment]
The main idea is solid, though is a bit buggy at the moment. Adding the `mode` option to the IVPLoss was a good idea. Also, given that torch.clip is an alias for torch.clamp, I will rework my `nearest_interpolation` function to include this to avoid an extra function call. 

Bugs:
1. There is a bug in the following line `init_values = torch.stack(x[-1::], dim=0).repeat(*(a,))`. The `x[-1::]` should be `x[::-1]`.
2. Also, the notation you use `*(r,)` in `x = torch.meshgrid(*(r,),indexing='ij')`, and in the example above, is redundant. You could just do `r` and avoid wrapping and unpacking it with `*(r,)`. 
3. The `_compute_init_values` method is a bit tricky. I built it specifically for the bilinear batched interpolation such that the init_values match the required shape for the function. The bilinear batched requires shape of (B, 2, H, W), but the documentation for your `nearest_interpolation_batched` function says (B, D, TYX) which would be inconsistent across the batched interpolation methods. Although, by chance or on purpose, I believe your function works for the (B, D, *shape) case assuming that D == len(shape), which should always be true I believe (although maybe this is what you meant by "TYX"?). I need to do some more testing here, but something will be needed here either on the `_compute_init_values`, documentation (if it generalizes trivially), or nearest batched interpolation side. Not sure which at the moment. 

Other than this, it looks good. I'll have a commit in by Wednesday hopefully with the updated headless opencv too.

**Comment by ryanirl** [comment]
I added your changes and fixed the bugs. Credit was given in the documentation for the `nearest_interpolation_batched` function. In my testing, using nearest neighbors interpolation gave a 1.25x speedup during training of an H1 model. That said, bilinear interpolation is still the default.

**Comment by kevinjohncutler** [comment]
Awesome! Much belated thanks for doing this.
